player.cpp

```diff
@@ -0,0 +1,28 @@
+#include "player.h"
+
+#include "gameobject.h"
+#include "input.h"
+#include "rectangle.h"
+
+#include <glm/glm.hpp>
+
+void Player::OnStart()
+{
+    mMesh = std::make_unique<Rectangle>(25.0f, 225.0f);
+    mPosition = glm::vec3(-550.0f, 0.0f, 0.0f);
+    mSpeed = 2.5f;
+}
+
+void Player::OnUpdate()
+{
+    if (Input::IsKeyPressed(GLFW_KEY_W))
+    {
+        mPosition.y += mSpeed;
+    }
+    else if (Input::IsKeyPressed(GLFW_KEY_S))
+    {
+        mPosition.y -= mSpeed;
+    }
+
+    std::cout << "Position x: " << mPosition.x << ", y: " << mPosition.y << std::endl;
```

Comment:
Could consider using a logging library to get this functionality and be able to tweak it, using debug/info logging:

https://github.com/gabime/spdlog

is a typical one I see recommended.

One thing to investigate for you. If this is being called in your main game loop, you may have concerns with realtime safety, similar to in audio code. If this call takes an undetermined amount of time to come back (because it's a system call) and you need to render a frame in the same callback, you may have funky frame drops or skippy frames. I'm not sure how most game engines deal with that, or if they are as paranoid as audio devs